src/commissioner/commissioner.cpp

```diff
@@ -354,13 +354,12 @@ void Commissioner::HandleCommissionerSet(const Coap::Message &aMessage, void *aC
             switch (state)
             {
             case Meshcop::kStateAccepted:
-                commissioner->mCommissionState = kStateAccepted;
                 break;
             case Meshcop::kStateRejected:
-                commissioner->mCommissionState = kStateRejected;
+                commissioner->mCommissionState = CommissionState::kStateRejected;
```

Comment:
This should not change the commissioner state. It only rejects the set request. The commissioner should continue to be active.